programs/marginfi/src/ix_utils.rs

```diff
@@ -53,10 +54,11 @@ pub fn validate_ix_first(
         let discrim = &instruction.data[0..8];
 
         // If this is the first non-compute ix, it is either the setup ix, or we fail
-        if !first_non_compute_ix_encountered {
+        if !start_ix_encountered {
             if instruction.program_id == *program_id && discrim == expected_hash {
-                first_non_compute_ix_encountered = true;
-            } else {
+                start_ix_encountered = true;
+            } else if !(instruction.program_id == *program_id && allowed_hashes.contains(&discrim))
```

Comment:
I think we also want to check that the program id is one of the two allowed (compute or kamino) otherwise there can be a hash collision with an ix from a different program